pkg/client-sdk/covenantless_client.go

```diff
@@ -2702,3 +2771,33 @@ func buildRedeemTx(
 
 	return bitcointree.BuildRedeemTx(ins, outs)
 }
+
+func handleOptions(options *SettleOptions) ([]*secp256k1.PrivateKey, []string, tree.SigningType, error) {
+	signerPrivKeys := make([]*secp256k1.PrivateKey, 0)
+	signerPubKeys := make([]string, 0)
+	if len(options.CustomSignerPrivKeys) > 0 {
+		for _, privKey := range options.CustomSignerPrivKeys {
+			signerPrivKeys = append(signerPrivKeys, privKey)
+			signerPubKeys = append(signerPubKeys, hex.EncodeToString(privKey.PubKey().SerializeCompressed()))
+		}
+	}
+
+	var signingType tree.SigningType
+	if options.SigningType != nil {
+		signingType = *options.SigningType
+	} else {
+		signingType = tree.SignBranch
+	}
+
+	// if no custom signer priv keys are provided, we generate a new ephemeral key
+	if len(signerPubKeys) == 0 {
+		ephemeralKey, err := secp256k1.GeneratePrivateKey()
+		if err != nil {
+			return nil, nil, tree.SignBranch, err
+		}
```

Comment:
maybe we should use an ephemeral key only when signing ALL, otherwise we should use a wallet key to be sure we can re-sign the branch if needed?

Reply:
in theory, sign ALL doesn't prevent the server to use the sweep-early feature so I'd keep wallet's key whatever the type